### ISSUE TITLE
Error in constrained species

### ISSUE THREAD

**Issue by tanzinaakther** [AtChem/AtChem2#485]
I am trying to run the model, but it says "constrained species" is not a species in the problem.  I have this specific species in the constrained file. Even if I delete that specific species, it gives the same error by saying about another species.
![image](https://user-images.githubusercontent.com/123010272/213258081-fc6cb67e-c25e-4e62-bb07-375556316963.png)

**Comment by rs028** [comment]
Can you check that:
1. `C2H2` is listed in `speciesConstrained.config`
2. A file named `C2H2` with the constrained data is present in `constraints/species`
3. `C2H2` is a variable in the `mechanism.fac` file

**Comment by tanzinaakther** [comment]
Thank you so much for the reply. C2H2 was not present in the mechanism.fac file but when I run it again, now it runs without any issue although I didnt add C2H2 in the mechanism file

**Comment by rs028** [comment]
Not sure why that is the case, but I suggest you add C2H2 to your mechanism file. Otherwise the results won't make any sense :)

**Comment by tanzinaakther** [comment]
Hi, I am getting this error again. This time, when I use TOLUENE, 124TMB and TBUT2ENE, It says these are not species in the problem. But when I removed this three, the model runs. I checked that these two species are present in mechanism file, constrained species. config and constraint/species

**Comment by rs028** [comment]
HI @tanzinaakther what do you mean "when I removed this three"?

**Comment by tanzinaakther** [comment]
I mean by deleting TOLUENE, 124TMB and TBUT2ENE, I do not get this error and the model runs

**Comment by rs028** [comment]
Deleting from the `speciesConstrained.config` file?

**Comment by tanzinaakther** [comment]
yes

**Comment by rs028** [comment]
If you remove them from the file, the model will not try to constrain them and so it will not complain that they are not part of the problem. That makes sense, but I don't understand why it fails if they are present in the chemical mechanism.
Are these species also present in `mechanism.f90`?

**Comment by tanzinaakther** [comment]
Hi, I see these species are not present in mechanism. f90 and mechanism.species. When I download mechanism file, and run it, should all species go into mechanism. f90 and mechanism.species? If not how can I add them in mechanism. f90 and mechanism.species?

**Comment by rs028** [comment]
This is odd. The mechanism.f90 file should be automatically generated when you run the build script.
Did you get any error at compilation time?

**Comment by tanzinaakther** [comment]
Let me get back to soon. I will try and inform you

**Comment by tanzinaakther** [comment]
Hi, I tried but faced the same problem. Please see attached.
<img width="776" alt="build run" src="https://user-images.githubusercontent.com/123010272/217686750-de1e6bb4-9c12-4de8-991f-4a901bd1ce16.PNG">
<img width="755" alt="run model" src="https://user-images.githubusercontent.com/123010272/217686753-656d5c4a-c2c7-462a-8475-76e549b3f99a.PNG">

**Comment by spco** [comment]
Notice the `AssertionError` in the first screenshot. This shows that the `./build/build_atchem2.sh` script has not completed successfully.

From the error, it looks as though you have either zero or >1 lines in `mechanism.fac` which contain the phrase `Generic Rate Coefficients`, which is causing an error. Can you please check this? If that seems fine, then the best bet at this point is to share your `mechanism.fac` here please so we can assist further.

**Comment by tanzinaakther** [comment]
Hi, I checked and see that the newly downloaded mecanism.fac file has nothing like  Generic Rate Coefficients. So, I copied this part from another mechanism.fac file and pasted it in my new mechanism.fac file. Also there were some breakage in the reactions line. One of the products created a 2nd line. So, I deleted them and made them in one line.

Now, everything is good. the model runs... Thank you. But my concern is why Generic rate constants were absent in newly downloaded mechanism file?
